src/providers/ModalsProvider.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useState,
+} from "react"
+import styles from "./ModalProvider.module.scss"
+
+interface ModalsContext {
+  content?: React.ReactNode
+  isOpen?: boolean
+  onPresent: (content: React.ReactNode, key?: string) => void
+  onClose: () => void
+}
+
+export const Context = createContext<ModalsContext>({
+  onPresent: () => undefined,
+  onClose: () => undefined,
+})
+
+const ModalsProvider: React.FC = ({ children }: PropsWithChildren<unknown>) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [content, setContent] = useState<React.ReactNode>()
+  const [, setModalKey] = useState<string>()
+
+  const handlePresent = useCallback(
+    (modalContent: React.ReactNode, key?: string) => {
+      setModalKey(key)
```

Comment:
What is `key` being used for here?

Reply:
I put this value to identify modal in the case of display modal on the another modal. But our case it doesn't require, so I removed.